### ISSUE TITLE
Gate PR's so they require CI checks to pass

### ISSUE THREAD

**Issue by jeffwidman** [python-zk/kazoo#698]
In https://github.com/python-zk/kazoo/pull/652, @StephenSorriaux very nicely fixed up our CI checks.

However, our current PR's aren't required to pass CI before being merged:
1. We should change this so they have to pass CI.
2. We need to retain the ability to override in case for some reason a PR intentionally won't pass CI but we still need to merge it. While very rare, this does occasionally happen. Several of the maintainers have admin rights on the repo so they could temporarily disable those branch checks if needed.

**Comment by a-ungurianu** [comment]
I agree with this, but I feel like fixing/ignoring the intermittent tests is a pre requisite for this

**Comment by ceache** [comment]
Agreed!

**Comment by ThosRTanner** [comment]
Even when the flaky tests are fixed, there still appears to be an issue with codecov, possibly related to https://github.com/codecov/codecov-action/issues/1855

**Comment by ceache** [comment]
Hopefully #824 sorts out the codecov issues. 
Test reliability should be addressed with #802 and all the other fixes I recently landed.

In any case, since our merges *are* gated on successful CI, I will close this issue and allow specific issues to be open for specific hindrance with our CI
